From 2a9fdd4ca690883922203646cd8ae2a36ddeeb80 Mon Sep 17 00:00:00 2001
From: Ryan Blue <blue@apache.org>
Date: Wed, 10 Jun 2015 20:24:09 -0700
Subject: [PATCH 078/140] CDK-1016: Fix OutputFormat writing directly to
 datasets.

This happens only when a dataset instance is passed to the configuration
methods. The fix is to verify that the target dataset is not the root by
inspecting whether the partition key has any values.
---
 .../data/mapreduce/DatasetKeyOutputFormat.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java b/kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java
index ade5052..637c972 100644
--- a/kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java
+++ b/kite-data/kite-data-mapreduce/src/main/java/org/kitesdk/data/mapreduce/DatasetKeyOutputFormat.java
@@ -461,7 +461,7 @@ public class DatasetKeyOutputFormat<E> extends OutputFormat<E, Void> {
       }
       FileSystemDataset fsDataset = (FileSystemDataset) target;
       PartitionKey key = fsDataset.keyFromDirectory(new Path(partitionDir));
-      if (key != null) {
+      if (key != null && !key.getValues().isEmpty()) {
         working = fsDataset.getPartition(key, true);
       }
       return new DatasetRecordWriter<E>(working);
-- 
1.7.9.5

